cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/common/DataprocUtils.java

```diff
@@ -229,6 +234,31 @@ public static String getSystemProjectId() {
     }
   }
 
+  /**
+   * Emit a dataproc metric.
+   **/
+  public static void emitMetric(ProvisionerContext context, String region,
+                                Exception e, String  metricName) {
```

Comment:
Annotate the `Exception e` with `@Nullable` since `null` is allowed

Reply:
Also, it is better to create another `emitMetric` method that doesn't take exception so that pass in `null` from the caller is not needed.

```
public static void emitMetrics(ProvieionerContext context, String region, String metricName) {
  emitMetrics(context, region, metricName, null);
}

public static void emitMetrics(ProvieionerContext context, String region, 
                               String metricName, @Nullable Exception e) {
  // Existing code ....
}

Reply:
done.